
# HG changeset patch
# User welpy-cw <h.w.forms@arcor.de>
# Date 1756138623 14400
# Node ID 61a9656dc5db3059afd5fa291f6e30c5f5c8c312
# Parent  de0f81aa90d9a8c19f86368ce3146d93739607c2
Bug 541744 - Do not mark NNTP messages read that could not be retrieved from the server. r=mkmelin


This had already been fixed in Bug 1757904, but has since regressed again.

Differential Revision: https://phabricator.services.mozilla.com/D259558

diff --git a/mail/base/content/msgHdrView.js b/mail/base/content/msgHdrView.js
--- a/mail/base/content/msgHdrView.js
+++ b/mail/base/content/msgHdrView.js
@@ -602,16 +602,26 @@ var messageProgressListener = {
    *
    * @param {nsIMailChannel} mailChannel
    * @see {nsIMailProgressListener}
    */
   onBodyComplete(mailChannel) {
     if (!this._channelIsCurrent(mailChannel)) {
       return;
     }
+
+    if (
+      mailChannel.URI instanceof Ci.nsIMsgMailNewsUrl &&
+      mailChannel.URI.errorCode
+    ) {
+      // This appears to be an NNTP message that couldn't be successfully
+      // retrieved from the server.
+      return;
+    }
+
     autoMarkAsRead();
   },
 
   /**
    * Step 4: The attachment information is available on the channel.
    *
    * @param {nsIMailChannel} mailChannel
    * @see {nsIMailProgressListener}

